flit/sdist.py

```diff
@@ -209,37 +210,44 @@ def build(self, target_dir:Path =None):
             target_dir.mkdir(parents=True)
         target = target_dir / '{}-{}.tar.gz'.format(
                         self.metadata.name, self.metadata.version)
-        tf = tarfile.open(str(target), mode='w:gz')
-        tf_dir = '{}-{}'.format(self.metadata.name, self.metadata.version)
-
-        files_to_add = self.find_tracked_files()
-
-        for relpath in files_to_add:
-            path = self.srcdir / relpath
-            tf.add(str(path), arcname=pjoin(tf_dir, relpath))
-
-        if 'setup.py' in files_to_add:
-            log.warning("Using setup.py from repository, not generating setup.py")
-        else:
-            setup_py = self.make_setup_py()
-            log.info("Writing generated setup.py")
-            ti = tarfile.TarInfo(pjoin(tf_dir, 'setup.py'))
-            ti.size = len(setup_py)
-            tf.addfile(ti, io.BytesIO(setup_py))
-
-        pkg_info = PKG_INFO.format(
-            name=self.metadata.name,
-            version=self.metadata.version,
-            summary=self.metadata.summary,
-            home_page=self.metadata.home_page,
-            author=self.metadata.author,
-            author_email=self.metadata.author_email,
-        ).encode('utf-8')
-        ti = tarfile.TarInfo(pjoin(tf_dir, 'PKG-INFO'))
-        ti.size = len(pkg_info)
-        tf.addfile(ti, io.BytesIO(pkg_info))
-
-        tf.close()
+        source_date_epoch = os.environ.get('SOURCE_DATE_EPOCH', '')
+        mtime = int(source_date_epoch) if source_date_epoch else None
+        gz = GzipFile(str(target), mode='wb', mtime=mtime)
+        tf = tarfile.TarFile(str(target), mode='w', fileobj=gz)
+
+        try:
+            tf_dir = '{}-{}'.format(self.metadata.name, self.metadata.version)
+
+            files_to_add = self.find_tracked_files()
```

Comment:
Is that guaranteed to return sorted files?

Reply:
Yup:

https://github.com/takluyver/flit/blob/8f5d6a71a5bc02ac0a0ba861059a5dd71d407829/flit/sdist.py#L165

Good thing to check, though :-)

Reply:
Nice. =) I am just thinking if explicitly sorting here makes the code more readable.